lib/Dialect/FIRRTL/Transforms/InferWidths.cpp

```diff
@@ -1082,6 +1082,11 @@ class InferenceMapping {
   /// non-aggregate.
   Expr *declareVar(FIRRTLType type, Location loc);
 
+  /// Assign the constraint expressions of the fields in the `result` argument
+  /// as the max of expressions in the `rhs` and `lhs` arguments. Both fields
+  /// must be the same type.
+  void maximumOfTypes(Value result, Value rhs, Value lhs);
```

Comment:
Better function name suggestion welcome.  Tried to match `constrainTypes` and `unifyTypes` but `maximizeTypes` doesn't sound right